pennylane/interfaces/tensorflow_autograph.py

```diff
@@ -361,34 +377,34 @@ def _execute(*all_params):  # pylint:disable=unused-argument
         output_sizes = res[-total_measurements:]
 
         if mode == "forward":
-            jacs = res[total_measurements:-total_measurements]
+            jacs = res[total_measurements * num_copies : -total_measurements]
 
-        res = res[:total_measurements]
+        res = res[: total_measurements * num_copies]

```

Comment:
You're right, it should be multiplied by `num_shot_copies`. I updated this.

If I understand correctly, we only use `mode="forward"` when the gradient function is a device method. Currently the only device method supported in `default.qubit` is adjoint (correct me if I'm wrong), and at the moment we don't support shot vectors with adjoint differentiation. This may be why this slipped the net.

Reply:
:+1: Agree with the points here.